src/client/app/effects/snippet.ts

```diff
@@ -187,6 +188,48 @@ export class SnippetEffects {
         .map((updatedSnippet) => new Snippet.SaveAction(updatedSnippet))
         .catch(exception => Observable.of(new UI.ReportErrorAction(Strings().snippetUpdateError, exception)));
 
+    @Effect({ dispatch: false })
+    openInPlayground$: Observable<Action> = this.actions$
+        .ofType(Snippet.SnippetActionTypes.OPEN_IN_PLAYGROUND)
+        .map(action => action.payload)
+        .map(payload => {
+            if (Utilities.platform === PlatformType.IOS) {
+                AI.trackEvent('Unsupported open in playground');
+                return;
+            }
+            let { type, id, isDownload } = payload;
+            let handler, extension;
+            switch (environment.current.host.toUpperCase()) {
+                case HostType.EXCEL:
+                    handler = 'ms-excel:ofe|u|';
+                    extension = '.xlsx';
+                    break;
+                case HostType.WORD:
+                    handler = 'ms-word:ofe|u|';
+                    extension = '.docx';
+                    break;
+                case HostType.POWERPOINT:
+                    case HostType.WORD:
+                    handler = 'ms-powerpoint:ofe|u|';
+                    extension = '.pptx';
+                    break;
+                default:
+                    return;
+            }
+            AI.trackEvent('Open in playground initiated');
+            let filename = `script-lab-playground-${environment.current.host}${extension}`;
```

Comment:
I was using this earlier, but I believe we discussed having a default filename (since the snippet is not going to be imported every time, the user might delete it or not run it, and in the case of importing the gist, the filename would be the id.docx, id.xlsx, etc, so not as pretty).